tests/LanguageServerRename/OffsetExtractor.php

```diff
@@ -82,20 +94,41 @@ public function parse(string $source): array
             }
 
             if (isset($this->rangeCloseMarkers[$result])) {
-                if (!isset($retVal[$this->rangeCloseMarkers[$result]])) {
-                    $retVal[$this->rangeCloseMarkers[$result]] = [];
-                }
-                $retVal[$this->rangeCloseMarkers[$result]][] = ByteOffsetRange::fromInts($currentRangeStartOffset, $offset); // $this->rangeCreateors[$result]($currentRangeStartOffset, $offset, $newSource);
+                $this->rangeResults[$this->rangeCloseMarkers[$result]][] = ByteOffsetRange::fromInts($currentRangeStartOffset, $offset);
                 continue;
             }
             
-
             $offset += strlen($result);
             $newSource .= $result;
         }
+        
+        $this->source = $newSource;
+
+        return $this;
+    }
+    /**
+     * @return ByteOffset[]
+     */
+    public function points(string $name): array
+    {
+        if(!isset($this->pointResults[$name]))
+            throw new RuntimeException('No registered point by that name.');
```

Comment:
This would be more helpful when developing: `sprintf('No point registered with name "%s", known names "%s"', $name, implode('", "', array_keys($this->pointResults)))`